### ISSUE TITLE
MP3 files erroneously classified as MPEG/MP1 or MPEG/MP2

### ISSUE THREAD

**Issue by nickcmaynard** [Borewit/music-metadata#681]
**Bug description**
I have some MP3 files which are (I believe) erroneously classified as MPEG/MP1, or MPEG/MP2 files.  Picard identifies these as "MPEG-1 Layer 3 - ID3v2.3"

**Expected behavior**
I expected them to be classified as MPEG/MP3 files.

**Audio file demonstrating the problem**
I can share these directly.

**Comment by Borewit** [comment]
`MPEG/MP1` is not something [music-metadata](https://github.com/Borewit/music-metadata) outputs. There is the container format and the coded format.

Container would be `MPEG`, and the codec something like `MPEG 1 Layer 3`. The container may be a bit artificial, to provide some consistency with other format. Also a rough indication which parser is used.

I have seen many examples of MPEG like files in which one frame says MPEG 1 Layer 1 and the next MPEG-1 Layer 3. If the first few frames are wrong... 

But maybe good an explicit unit tests at least demonstrating the differences between these formats.

Some sample files: http://fileformats.archiveteam.org/wiki/MPEG_Audio_Layer_I

**Comment by nickcmaynard** [comment]
Ah yes, apologies - I was unclear.

An example:

```js
{
  codec: "MP1",
​  codecProfile: "V10",
​  container: "MPEG",
​  duration: 280.08489795918365,
​  lossless: false,
​  numberOfChannels: 2,
​  sampleRate: 48000,
​  tagTypes: [ "ID3v2.3", "ID3v1" ]
}
```

The file in question is a 44100Hz MP3.  This is with music-metadata v4.6.0.

**Comment by Borewit** [comment]
Version [v4.6.0](https://github.com/Borewit/music-metadata/releases/tag/v4.6.0)? Why don't you update first.

**Comment by nickcmaynard** [comment]
No problem! 

On v5.4.3, same problem I'm afraid, just with the enhanced MP3 codec descriptions from #350.

```js
{
  codec: "MPEG 1 Layer 1",
​  codecProfile: "V10",
​  container: "MPEG",
​  duration: 280.08489795918365,
  lossless: false,
  numberOfChannels: 2,
  sampleRate: 48000,
  tagTypes: [ "ID3v2.3", "ID3v1" ]
​}
```

**Comment by Borewit** [comment]
Most of the time we, developers, just want confirmation that the issue is on the latest version. So we know:
-  the issue is relevant
- we are talking about the same version 
- if there is a a fix, it will be done one the latest version 

Fixing an old version is not a desired way of working, that is only done if there is a good justification.

Can you please provide me with a sample file @nickcmaynard?

**Comment by nickcmaynard** [comment]
Sent via email - please let me know if you have not received it...

**Comment by nickcmaynard** [comment]
... also checked w/ v7.5.0.

**Comment by Borewit** [comment]
> Sent via email - please let me know if you have not received it...

Got it, thx.

It looks like this caused by a bug in [music-metadata](https://github.com/Borewit/music-metadata). It looks like the file has a shorter Xing header then expected, causing a synchronization error. The "zone B" as described [here](http://gabriel.mp3-tech.org/mp3infotag.html) is not present in this file.

**Comment by nickcmaynard** [comment]
Thank you. I do have some other examples which show as different codecs (old MP2 etc), which I could provide to make sure the fix is complete. Would that be helpful?

**Comment by Borewit** [comment]
> Thank you. I do have some other examples which show as different codecs (old MP2 etc), which I could provide to make sure the fix is complete. Would that be helpful?

That may be useful indeed.

If you can shrink the samples, that would be really helpful. I take the first 10 MPEG frames of originals, to keep the text fixture small and to make them copyright free.

e.g. (where `70827` is the offset of frame #10):

```shell script
#!/bin/sh

IN="05 Solace.mp3"
OUT="Solace.mp3"

dd skip=0 count=70827 bs=1 "if=$IN" "of=$OUT"

FILESIZE=$(stat -c%s "$IN")

echo "FILESIZE=$FILESIZE"

dd skip=$((FILESIZE - 128)) count=128 bs=1 "if=$IN" of=id3v1

cat id3v1 >> "$OUT"
```

**Comment by nickcmaynard** [comment]
No problem - will do.  Apologies for the (potentially) silly question, but I'm unfamiliar with the MPEG data structure - do you have a method for finding the offset?

FWIW, #683 does appear to fix the cases I have, but I'd like to help make sure you have a full regression suite...

**Comment by Borewit** [comment]
There are 2 ways I use to figure out the offset:

**Using [MPEG-Audio-Info 2.2](http://www.softsea.com/download/MPEG-Audio-Info.html)**

![image](https://user-images.githubusercontent.com/23255260/100545788-e0cfe380-325d-11eb-8ebf-e09147bcaafa.png)

**Using music-metadata**:
Defined, set `DEBUG=-mocha:*,*` as an environment variable.

Change value `3` to `11` in: https://github.com/Borewit/music-metadata/blob/56c0efa6136659a7896548b36409753bcc74a11e/lib/mpeg/MpegParser.ts#L467 

Check the log for ` Sync at offset`:
```log
music-metadata:parser:mpeg Sync at offset=70827, frameCount=10
```

Note that [music-metadata](https://github.com/Borewit/music-metadata) starts to count at 0, which is maybe a bit odd.

**Comment by Borewit** [comment]
Note that I added a list of MPEG 1 Layer I, II & III tests in #683.

**Comment by Borewit** [comment]
Fixed in [v7.5.1](https://github.com/Borewit/music-metadata/releases/tag/v7.5.1)
